collections_hierarchy/mixins.py

```diff
@@ -1,54 +1,81 @@
+from copy import deepcopy
+
 class ComparableMixin(object):
+    #Implements comparison functionality. Common for List and Dict
     def __eq__(self, other):
-        pass
+        data1 = getattr(self, self.DATA_ATTR_NAME) #Get the value of the property that stores the data
+        data2 = getattr(other, other.DATA_ATTR_NAME)
+        
+        if isinstance(data1, dict):
+            return data1 == data2
```

Comment:
Also unless I'm missing something here should it not be:
```python
return sorted(data1) == sorted(data2)
```
for dicts and
```python
return data1 == data2
```
for the list (the other way around to what you currently have).